src/sinks/util/mod.rs

```diff
@@ -104,14 +106,315 @@ impl<I> EncodedEvent<I> {
+                                "LOCAL0" => 16,
+                                "LOCAL1" => 17,
+                                "LOCAL2" => 18,
+                                "LOCAL3" => 19,
+                                "LOCAL4" => 20,
+                                "LOCAL5" => 21,
+                                "LOCAL6" => 22,
+                                "LOCAL7" => 23,
+                                _ => 24,
+                            };
+                            if num > 23 {
+                                return 1 // USER
+                            } else {
+                                return num
+                            }
+                        }
+                    }
+            } else {
+                return 1 // USER
+            }
+        }
+    }
+}
+
+fn get_num_severity(config_severity: &Severity, log: &LogEvent) -> u8 {
+    match config_severity {
+        Severity::Fixed(num) => return *num,
+        Severity::Field(field_name) => {
+            if let Some(field_value) = log.get(field_name.as_str()) {
+                let field_value_string = String::from_utf8(field_value.coerce_to_bytes().to_vec()).unwrap_or_default();
+                let num_value = field_value_string.parse::<u8>();
+                match num_value {
+                    Ok(num) => {
+                        if num > 6 {
+                            return 5 // INFORMATIONAL
+                        } else {
+                            return num
+                        }
+                    }
+                    Err(_) => {
+                            let num = match field_value_string.to_uppercase().as_str() {
+                                "EMERGENCY" => 0,
+                                "ALERT" => 1,
+                                "CRITICAL" => 2,
+                                "ERROR" => 3,
+                                "WARNING" => 4,
+                                "INFORMATIONAL" => 5,
+                                "DEBUG" => 6,
+                                _ => 7,
+                            };
+                            if num > 6 {
+                                return 5 // INFORMATIONAL
+                            } else {
+                                return num
+                            }
+                        }
+                    }
+            } else {
+                return 5 // INFORMATIONAL
+            }
+        }
+    }
+}
+
+fn get_field(config_name: &String, log: &LogEvent) -> String {
+    if let Some(field_name) = config_name.strip_prefix("$.message.") {
+        if let Some(field_value) = log.get(field_name) {
+            return String::from_utf8(field_value.coerce_to_bytes().to_vec()).unwrap_or_default();
+        } else {
+            return "no_such_field".to_string()
+        }
+    } else {
+        return config_name.clone()
+    }
+}
+
+fn get_timestamp(log: &LogEvent) -> DateTime::<Local> {
+    match log.get("@timestamp") {
```

Comment:
is this specific to our implementation needs for openshift?

Reply:
vector defines a timestamp key as part of its log schema in `src/config/log_schema.rs` we can use that. 
```
    pub fn timestamp_key(&self) -> &str {
        &self.timestamp_key
    }
```

Reply:
This PR is not going upstream as it is, there are things in it that don't make sense outside of our use case.